File: Question_01.js
// Import the 'readline' module to handle input and output
const readline = require('readline');

// Create an interface for input and output using stdin and stdout
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

// Prompt the user to enter an integer N
rl.question('Enter an integer (N): ', (N) => {
  // Convert N to an integer
  N = parseInt(N);
  
  // Check if value of N is not a valid number.
  // Conditional check to determine if N is less than or equal to 0
  if (isNaN(N) || N <= 0) {
    console.log("Invalid input. N should be a positive integer.");
    rl.close();
    return;
  }

  const inputSets = [];

  // Function to read input for a single set
  function readInputSet(index) {
    // Prompt the user to enter a value for M of the current set
    rl.question(`Enter value for M-${index + 1}: `, (M) => {
      const num = parseInt(M);

      if (!isNaN(num)) {
        inputSets.push(num);

        if (inputSets.length === N) {
          // Print the conditions for each input value
          inputSets.forEach((input) => {
            let output = '';

            // Check if the input is a multiple of 3 and/or 5
            if (input % 3 === 0) {
              output += "Ona ";
            }
            if (input % 5 === 0) {
              output += "Data ";
            }
            if (input % 3 === 0 && input % 5 === 0) {
              output += "OnaData ";
            }
            if (output === '') {
              output = "N/A";
            }

            // Display the conditions for the input value
            console.log(`Conditions for input ${input}: ${output}`);
          });

          // Close the input interface
          rl.close();
        } else {
          // Ask for the next input set
          readInputSet(index + 1);
        }
      } else {
        console.log("Invalid input. Please enter an integer.");
        // Retry input for the same set
        readInputSet(index);
      }
    });
  }

  // Start reading input sets from index 0
  readInputSet(0);
});

File: question_01.py
# Read the number of lines
N = int(input())

# Read M numbers and store them in a list
numbers = []
# loop iterates 'N' times, where 'N' is the number of lines specified by the user.
for _ in range(N):
    # Read an integer input from the user. This input represents the value 'M'.
    M = int(input())
    # Append the value 'M' to the 'numbers' list, which stores the input integers.
    numbers.append(M)

# Process and output the result for each number
for M in numbers:
    found = False  # Initialize a flag to track if any divisible number is found
    result = ""  # Initialize a string to store the result for the current number

    for num in range(1, M + 1):
        if num % 3 == 0 and num % 5 == 0:
            # If divisible by both 3 and 5, add "OnaData" to the result
            result += "OnaData "
            found = True
        elif num % 3 == 0:
            # If divisible by 3, add "Ona" to the result
            result += "Ona "
            found = True
        elif num % 5 == 0:
            # If divisible by 5, add "Data" to the result
            result += "Data "
            found = True

    if not found:
        # If no divisibility condition is met, add "N/A" to the result
        result += "N/A "

    # Print the result for the current number after removing trailing space
    print(result.strip())
